### PR TITLE
[ci] Divide .gitlab-ci.yml into several files

### PR DESCRIPTION
**Pull request by alvicsam**
Attempt to divide `.gitlab-ci.yml` into several files. 
Part of https://github.com/paritytech/ci_cd/issues/325

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -50,49 +50,50 @@ default:
...
+    - cargo --version
+    - rustup +nightly show
+    - cargo +nightly --version
 
-.docker-env:                       &docker-env
+.docker-env:
   image:                           "${CI_IMAGE}"
   before_script:
-    - *rust-info-script
+    - !reference [.rust-info-script, script]
```
Comment (TriplEight) [inline .gitlab-ci.yml:84]: this looks a lil' bit ugly for me, but anchors can't be used across files.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -140,35 +141,17 @@ default:
...
-    | tee ./artifacts/benches/$CI_COMMIT_REF_NAME-$CI_COMMIT_SHORT_SHA/::trie::read::small.json'
-  - sccache -s
-
-
 #### stage:                       .pre
 
 skip-if-draft:
-  image:                           paritytech/tools:latest
-  <<:                              *kubernetes-env
+  extends:                         .kubernetes-env
```
Comment (TriplEight) [inline .gitlab-ci.yml:152]: `extends:` has to be used instead of `<<: *anchor-name` notation.
Just documenting here.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -178,568 +161,12 @@ skip-if-draft:
...
-    - cargo deny check -c ./scripts/ci/deny.toml 2> deny.log
-  artifacts:
-    name:                          $CI_COMMIT_SHORT_SHA
-    expire_in:                     3 days
-    when:                          always
-    paths:
-      - deny.log
-  # FIXME: Temporarily allow to fail.
-  allow_failure:                   true
+include:
```
Comment (TriplEight) [inline .gitlab-ci.yml:164]: Please write on top of this file that it's been divided into these ones.
And also please add the description on top of each of them.

**Review by TriplEight — CHANGES_REQUESTED** [review]

**Comment by alvicsam** [comment]
Merge after https://github.com/paritytech/substrate/pull/11384

**Review by sergejparity — APPROVED** [review]

**Pull request merged**